src/app/modules/item/services/item-datasource.service.ts

```diff
@@ -22,11 +22,15 @@ export interface ItemData { route: FullItemRoute, item: Item, breadcrumbs: Bread
 @Injectable()
 export class ItemDataSource implements OnDestroy {
 
+  private readonly fetchOperation$ = new ReplaySubject<FullItemRoute>(1); // trigger item fetching
+  private readonly refresh$ = new Subject<void>();
+
   /* state to put outputted */
-  private state = new BehaviorSubject<FetchState<ItemData>>(fetchingState());
-  state$ = this.state.asObservable();
+  readonly state$ = this.fetchOperation$.pipe(
+    switchMap(item => this.fetchItemData(item)),
+    mapToFetchState({ resetter: merge(this.refresh$, this.fetchOperation$) })
```

Comment:
Why do you need the `fetchOperation$` as resetter here ?
To me it's useless since the `fetchOperation$` subject is already the starting point of the observable, did you encounter issues when not providing it as resetter ?

Reply:
Otherwise, on navigation, the former item is displayed while fetching the next one. Actually, that was the initial goal of the resetter, more than for refreshing (otherwise it would have been called reloader/refresher), but at the end it has not been used for this usage yet. If you think about it, the `mapToFetchState` has no other way the `this.fetchOperation$` has changed... it just knows the output of the `switchMap`, which comes with a delay.